scipy/spatial/spherical_voronoi.py

```diff
@@ -0,0 +1,346 @@
+    center : double array of shape (3,)
+            center of the sphere
+            Default: None (assumes sphere is centered at origin)
+    vertices : double array of shape (nvertices, 3)
+            Voronoi vertices corresponding to points
+    regions : list of list of integers of shape (npoints, _ )
+            the n-th entry is a list consisting of the indices
+            of the vertices belonging to the n-th point in points
+
+    Notes
+    ----------
+    The spherical Voronoi diagram algorithm proceeds as follows. The Convex
+    Hull of the input points (generators) is calculated, and is equivalent to
+    their Delaunay triangulation on the surface of the sphere [Caroli]_.
+    A 3D Delaunay tetrahedralization is obtained by including the origin of
+    the coordinate system as the fourth vertex of each simplex of the Convex
+    Hull. The circumcenters of all tetrahedra in the system are calculated and
+    projected to the surface of the sphere, producing the Voronoi vertices.
+    The Delaunay tetrahedralization neighbour information is then used to
+    order the Voronoi region vertices around each generator. The latter
+    approach is substantially less sensitive to floating point issues than
+    angle-based methods of Voronoi region vertex sorting.
+
+    The surface area of spherical polygons is calculated by decomposing them
+    into triangles and using L'Huilier's Theorem to calculate the spherical
+    excess of each triangle [Weisstein]_. The sum of the spherical excesses is
+    multiplied by the square of the sphere radius to obtain the surface area
+    of the spherical polygon. For nearly-degenerate spherical polygons an area
+    of approximately 0 is returned by default, rather than attempting the
+    unstable calculation.
+
+    Empirical assessment of spherical Voronoi algorithm performance suggests
+    quadratic time complexity (loglinear is optimal, but algorithms are more
+    challenging to implement). The reconstitution of the surface area of the
+    sphere, measured as the sum of the surface areas of all Voronoi regions,
+    is closest to 100 % for larger (>> 10) numbers of generators.
+
+    References
+    ----------
+
+    .. [Caroli] Caroli et al. Robust and Efficient Delaunay triangulations of
+                points on or close to a sphere. Research Report RR-7004, 2009.
+    .. [Weisstein] "L'Huilier's Theorem." From MathWorld -- A Wolfram Web
+                Resource. http://mathworld.wolfram.com/LHuiliersTheorem.html
+
+    See Also
+    --------
+    Voronoi : Conventional Voronoi diagrams in N dimensions.
+
+    Examples
+    --------
+
+    >>> from matplotlib import colors
+    >>> from mpl_toolkits.mplot3d.art3d import Poly3DCollection
+    >>> import numpy as np
+    >>> import matplotlib.pyplot as plt
+    >>> from scipy.spatial import spherical_voronoi
+    >>> from mpl_toolkits.mplot3d import proj3d
+    >>> import scipy as sp
```

Comment:
this import can be deleted as well, it does nothing and ``scipy as sp`` should not be used anymore for anything